cvxpy/constraints/nonpos.py

```
@@ -156,7 +180,6 @@ class Inequality(Constraint):
     """
     def __init__(self, lhs, rhs, constr_id=None) -> None:
         self._expr = lhs - rhs
-        # TODO remove this restriction.
```

I removed this TODO since I don't think adding this "feature" would actually be a good idea.

---

To explain my position here, Python doesn't allow inequality-based comparisons between complex numbers. MATLAB allows it but disregards the imaginary part when making the comparison. There's no viable course of action where we can allow complex-valued inequality constraints and not end up confusing the heck out of some people. It's better not to support this and fail loudly than to risk hidden bugs for a tiny bit of syntactic sugar.